### PR TITLE
RTM 1.1 of pw/email reset

### PR DESCRIPTION
**Pull request by Rick-Anderson**
[Review URL](https://review.docs.microsoft.com/en-us/aspnet/core/security/authentication/accconfirm?branch=ra%2Fpassword-email-sendgrid)

### CODE REVIEW

File: aspnetcore/security/authentication/accconfirm.md [hunk 1]
```diff
@@ -86,128 +74,113 @@ You generally want to prevent new users from posting any data to your web site b
 
 Update `ConfigureServices` to require a confirmed email:
 
-[!code-csharp[Main](accconfirm/sample/WebApplication3/src/WebApplication3/Startup.cs?highlight=11&name=snippet1)]
+[!code-csharp[Main](accconfirm/sample/WebApp1/Startup.cs?name=snippet1&highlight=13-16)]
 
 ### Configure email provider
 
-We'll use the [Options pattern](../../fundamentals/configuration.md#options-config-objects) to access the user account and key settings. For more information, see [configuration](../../fundamentals/configuration.md#fundamentals-configuration).
+We'll use the [Options pattern](xref:fundamentals/configuration#options-config-objects) to access the user account and key settings. For more information, see [configuration](xref:fundamentals/configuration#fundamentals-configuration).
```
Comment (tdykstra) [inline aspnetcore/security/authentication/accconfirm.md]: Add intro text explaining you're going to use SendGrid to send emails, and they'll need a SendGrid account.  That's the account and key this is talking about.

File: aspnetcore/security/authentication/accconfirm.md [hunk 2]
```diff
@@ -86,128 +74,113 @@ You generally want to prevent new users from posting any data to your web site b
...
+  ```json
+  {
+    "SendGridUser": "RickAndMSFT",
+    "SendGridKey": "<key removed>"
+  }
+  ```
 
 ### Configure startup to use `AuthMessageSenderOptions`
 
 Add the dependency `Microsoft.Extensions.Options.ConfigurationExtensions` in the project.json file.
```
Comment (tdykstra) [inline aspnetcore/security/authentication/accconfirm.md]: Note project.json ref

File: aspnetcore/security/authentication/accconfirm.md [hunk 3]
```diff
@@ -86,128 +74,113 @@ You generally want to prevent new users from posting any data to your web site b
...
 
-[!code-csharp[Main](../../security/authentication/accconfirm/sample/WebApplication3/src/WebApplication3/Startup.cs?highlight=4&range=62-65)]
+[!code-csharp[Main](accconfirm/sample/WebApp1/Startup.cs?name=snippet1&highlight=26)]
 
-### Configure the `AuthMessageSender` class
+### Configure the AuthMessageSender class
 
 This tutorial shows how to add email notification through [SendGrid](https://sendgrid.com/), but you can send email using SMTP and other mechanisms.
 
 * Install the SendGrid.NetCore NuGet package. From the Package Manager Console,  enter the following the following command:
```
Comment (tdykstra) [inline aspnetcore/security/authentication/accconfirm.md]: * Format package name
* The name here is out of sync with the following command line example

File: aspnetcore/security/authentication/accconfirm.md [hunk 4]
```diff
@@ -86,128 +74,113 @@ You generally want to prevent new users from posting any data to your web site b
...
 ## Enable account confirmation and password recovery
 
 The template already has the code for account confirmation and password recovery. Follow these steps to enable it:
 
 *  Find the `[HttpPost] Register` method in the  *AccountController.cs* file.
-
 *  Uncomment the code to enable account confirmation.
 
-[!code-csharp[Main](accconfirm/sample/WebApplication3/src/WebApplication3/Controllers/AccountController.cs?highlight=17-21&range=113-142)]
+[!code-csharp[Main](accconfirm/sample/WebApp1/Controllers/AccountController.cs?highlight=19-25&name=snippet_Register)]
```
Comment (tdykstra) [inline aspnetcore/security/authentication/accconfirm.md]: 3 lines above the highlighted lines need to be highlighted.

File: aspnetcore/security/authentication/accconfirm.md [hunk 5]
```diff
@@ -86,128 +74,113 @@ You generally want to prevent new users from posting any data to your web site b
...
 
-> [!NOTE]
-> We're also preventing a newly registered user from being automatically logged on by commenting out the following line:
+Note: We're also preventing a newly registered user from being automatically logged on by commenting out the following line:
 >
```
Comment (tdykstra) [inline aspnetcore/security/authentication/accconfirm.md]: why do you have the angle brackets here?

File: aspnetcore/security/authentication/accconfirm.md [hunk 6]
```diff
@@ -86,128 +74,113 @@ You generally want to prevent new users from posting any data to your web site b
...
 
 ## Register, confirm email, and reset password
 
 In this section, run the web app and show the account confirmation and password recovery flow.
 
-* Run the application and register a new user
+* Run the app and register a new user
 
-![Web application Account Register view](accconfirm/_static/loginaccconfirm1.png)
+ ![Web application Account Register view](accconfirm/_static/loginaccconfirm1.png)
```
Comment (tdykstra) [inline aspnetcore/security/authentication/accconfirm.md]: Note certificate error

File: aspnetcore/security/authentication/accconfirm.md [hunk 7]
```diff
@@ -86,128 +74,113 @@ You generally want to prevent new users from posting any data to your web site b
...
-
+* If you're logged in, select **Logout**.  
+* Select the **Log in** link and select the **Forgot your password?** link.
 * Enter the email you used to register the account.
-
 * An email with a link to reset your password will be sent. Check your email and click the link to reset your password.  After your password has been successfully reset, you can login with your email and new password.
 
 ## Require email confirmation before login
 
 With the current templates, once a user completes the registration form, they are logged in (authenticated). You generally want to confirm their email before logging them in. In the section below, we will modify the code to require new users have a confirmed email before they are logged in. Update the `[HttpPost] Login` action in the *AccountController.cs* file with the following highlighted changes.
```
Comment (tdykstra) [inline aspnetcore/security/authentication/accconfirm.md]: This is confusing -- I thought we just did this (required email confirmation to log in) and verified that it worked.

Comment (Rick-Anderson) [inline aspnetcore/security/authentication/accconfirm.md]: You're right. What this is really doing is not logging in a user once they register.

Comment (Rick-Anderson) [inline aspnetcore/security/authentication/accconfirm.md]: changed to ## Prevent login at registration

File: aspnetcore/security/authentication/accconfirm.md (hunk 6 above)
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/accconfirm.md]: I'm not seeing a cert error in this png or any other?

**Comment by Rick-Anderson** [comment]
>why do you have the angle brackets here?
I'm guessing it's an artifact of the conversion process. I didn't touch that line or look at it. Removed them anyway. Good catch.

Not seeing cert error.

**Pull request merged**